student-info-system/professor/views.py

```diff
@@ -47,3 +58,37 @@ def student(request, studentid):
     ssects = stud.sectionstudent_set.all()
     data = {'student': stud, 'ssects': ssects}
     return render(request, 'professor/student.html', data)
+
+
+@role_login_required(Profile.ACCESS_PROFESSOR)
+def add_reference(request, sectionid):
+    data = {'sectionid': sectionid}
```

Comment:
why do you go through section first? Why not index on courses in major?

Reply:
wrote this before I knew it was by course